module/zfs/zfs_znode.c

```diff
@@ -492,6 +492,8 @@ zfs_inode_set_ops(zfs_sb_t *zsb, struct inode *ip)
 	}
 }
 
+static void zfs_inode_update_new(znode_t *zp);
```

Comment:
As a general rule forward declarations are discouraged if it's easy to reorder the functions to avoid it.  Please go ahead and just move the `zfs_inode_update_impl` function up and wrappers up.